File: Python/4. 큐/9093.py
'''
 * Author : JC
 * https://www.acmicpc.net/problem/9093
'''

t = int(input())

for i in range(t) :
    s = input().split()
    l = []
    for j in s : 
        l.append(''.join(reversed(j)))
    # print(*l) -> [Line 14 ~ 15] 코드를 이렇게 한 줄로 작성해도 된다.
    for i in l :
        print(i, end = ' ')

File: C++/4.1 큐 - 초급/9093.cpp
/*
 * Author : JC
 * https://www.acmicpc.net/problem/9093
*/

#include <iostream>
#include <queue>
#include <algorithm>

using namespace std;

int n;
string str;
queue<string> q;

int main(){
    
    cin >> n;
    cin.ignore();
    for(int i=0; i<n; i++){
        getline(cin, str);
        str += " ";
        int len = str.length();
        string tmp;
        for(int j=0; j<len; j++){
            if(str[j] == ' '){
                reverse(tmp.begin(), tmp.end());
                q.push(tmp);
                tmp.clear();
            }else{
                tmp += str[j];
            }
        }
        while(!q.empty()){
            cout << q.front() << " ";
            q.pop();
        }
        cout << "\n";
    }

    return 0;
}